datafusion-cli/src/object_storage.rs

```diff
@@ -493,29 +576,27 @@ mod tests {
         );
 
         let ctx = SessionContext::new();
```

Comment:
Looking at this with whitespace blind diff https://github.com/apache/datafusion/pull/16300/files?w=1 

makes it easier to see what is happening

Specifically, I just refactored the parsing / table_options extraction into a separate function to reduce repetition.

Otherwise the existing tests are the same